spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/reactive/KubernetesInformerReactiveDiscoveryClientAutoConfiguration.java

```diff
@@ -95,16 +100,25 @@ public KubernetesInformerReactiveDiscoveryClient kubernetesReactiveDiscoveryClie
 				serviceLister, endpointsLister, serviceInformer, endpointsInformer, properties);
 	}
 
+	@Deprecated(forRemoval = true)
+	KubernetesDiscoveryClientHealthIndicatorInitializer reactiveIndicatorInitializer(
+			ApplicationEventPublisher applicationEventPublisher, PodUtils<?> podUtils) {
+		LOG.debug(() -> "Will publish InstanceRegisteredEvent from reactive implementation");
+		return new KubernetesDiscoveryClientHealthIndicatorInitializer(podUtils, applicationEventPublisher);
+	}
+
 	/**
 	 * Post an event so that health indicator is initialized.
 	 */
 	@Bean
 	@ConditionalOnClass(name = "org.springframework.boot.actuate.health.ReactiveHealthIndicator")
 	@ConditionalOnDiscoveryHealthIndicatorEnabled
 	KubernetesDiscoveryClientHealthIndicatorInitializer reactiveIndicatorInitializer(
-			ApplicationEventPublisher applicationEventPublisher, PodUtils<?> podUtils) {
```

Comment:
Github is doing weird stuff with the diff, it looks like you added a deprecated method ;)

I am curious as to why the IT was failing with the injected PodUtils?

Reply:
Actually I am rereading the issue and now I understand why.

However I am not sure I agree with this, now that I think about it.

I will comment on the issue since that has more context as to the problem at hand